pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -477,13 +516,47 @@ stdenv.mkDerivation ({
     # TODO: fetch the self from the fixpoint instead
     haddockDir = self: if doHaddock then "${docdir self.doc}/html" else null;
 
-    env = shellFor {
-      packages = p: [ drv ];
-      inherit shellHook;
-    };
-
+    envFunc = { withHoogle ? false, ... } @ args:
```

Comment:
Could you add a comment about what this function does (and what types of arguments it is expecting and what it returns)?

Reply:
Documentation added in newest changes